publication-commons/src/test/java/no/unit/nva/publication/service/impl/ResourceServiceTest.java

```diff
@@ -1187,6 +1187,18 @@ void shouldThrowBadRequestWhenAttemptingToDeletePublishedPublication() throws Ap
                      () -> resourceService.deletePublication(resourceService.getPublication(publication)));
     }
 
+    @Test
+    void shouldUpdatePublicationByImportedEntry() throws ApiGatewayException {
+        var publication = createPublishedResource();
+        publication.setDoi(randomDoi());
+        var updatedPublication = resourceService.updatePublicationByImportEntry(publication,
+                                                                   ImportSource.fromSource(Source.CRISTIN));
+
+        assertThat(updatedPublication.getImportDetails().getFirst().importSource().getSource(),
```

Comment:
Add an assertion that `getImportDetails().size == 1`? To make sure we're not call the method too many times

Reply:
Or maybe it's better to assert that in the handler tests

Reply:
Will assert in both of tests